### PR TITLE
feat(etcd-backup): add custom CA support

### PR DESCRIPTION
**Pull request by vmaillot**
Title says it all.

### CODE REVIEW

File: backup.sh [hunk 1]
```diff
@@ -35,8 +35,11 @@ set -xeuo pipefail
 if [ "${OCP_BACKUP_S3}" = "true" ]; then
     # prepare & push backup to S3
 
-    # update CA trust
-    update-ca-trust
+    # add custom CA if any and update CA trust
+    if [ "${OCP_BACKUP_S3_CA}" ]; then
+        echo -n "${OCP_BACKUP_S3_CA}" > /etc/pki/ca-trust/source/anchors/ca.crt
+        update-ca-trust
+    fi
```
Comment (hairmare) [inline backup.sh:42]: why can't we just mount the ca to `/etc/pki/ca-trust/source/anchors/ca.crt` from a ConfigMap?

File: README.md [hunk 2]
```diff
@@ -62,6 +62,7 @@ The following options are used:
 - `OCP_BACKUP_S3`: Use S3 to store etcd-backup snapshots
 - `OCP_BACKUP_S3_NAME`: MinIO client host alias name
 - `OCP_BACKUP_S3_HOST`: S3 host endpoint (with scheme)
+- `OCP_BACKUP-S3_CA`: S3 host CA (if needed)
```
Comment (inisitijitty) [inline README.md:65]:
```suggestion
- `OCP_BACKUP_S3_CA`: S3 host CA (if needed)
```

File: backup.sh (hunk 1 above)
Comment (vmaillot) [reply backup.sh:42]: this will imply having a dedicated configMap for just the CA, I wanted to avoid that, what do you think?

Comment (hairmare) [reply backup.sh:42]: afair that's what the previous `update-ca-trust` call is for

Comment (vmaillot) [reply backup.sh]: what about this @hairmare?

Comment (hairmare) [reply backup.sh]: afaict we can create configure the cluster to know about the CA and create an empty configmap with the `config.openshift.io/inject-trusted-cabundle="true"` annotation using this chaart and then magick should happen:

https://github.com/openshift/openshift-docs/blob/main/modules/certificate-injection-using-operators.adoc

Comment (hairmare) [reply backup.sh]: Specifically, the cluster *should* already know about the right CA. We use the same trust root that for basics like accessing internal registries. it is usually configured during installation, moreso in air-gapped cases than others.

The certificate injection feature seemingly aims to cover the same need that [trust-manager](https://cert-manager.io/docs/trust/trust-manager/) wants to cover, but as a CNCF (sub-)project. The [openshift/service-ca-operator](https://github.com/openshift/service-ca-operator) seems to be RedHat's upstream for their implementation that most likely predates trust-manager.

**Comment by vmaillot** [comment]
i am closing this PR as we should be able to that in the helm chart only.

**Pull request closed without merging**